sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -0,0 +1,164 @@
+
+### Serialization
+
+```C# Snippet:NewtonSoft_Serialize
+DogListProperty dog = new DogListProperty
+{
+    Name = "Doggo",
+    IsHungry = true,
+    Weight = 1.1,
+    FoodConsumed = { "kibble", "egg", "peanut butter" },
+};
+ModelSerializerOptions options = new ModelSerializerOptions();
+options.GenericTypeSerializerCreator = type => type.Equals(typeof(DogListProperty)) ? new NewtonsoftJsonObjectSerializer() : null;
+
+BinaryData data = ModelSerializer.Serialize(dog, options);
+```
+
+### Deserialization
+
+```C# Snippet:NewtonSoft_Deserialize
+ModelSerializerOptions options = new ModelSerializerOptions();
+options.GenericTypeSerializerCreator = type => type.Equals(typeof(DogListProperty)) ? new NewtonsoftJsonObjectSerializer() : null;
+string json = @"[{""LatinName"":""Animalia"",""Weight"":1.1,""Name"":""Doggo"",""IsHungry"":false,""FoodConsumed"":[""kibble"",""egg"",""peanut butter""],""NumberOfLegs"":4}]";
+
+DogListProperty dog = ModelSerializer.Deserialize<DogListProperty>(BinaryData.FromString(json), options);
+```
+
+## Using ModelJsonConverter for JsonSerializer
+
+In order to better integrate with the rest of the .NET ecosystem, Azure.Core supports System.Text.Json serialization. The following example demonstrates using System.Text.Json for serialization and deserialization
+If we go this route the IJsonSerializable interface will only be needed for compile time constraints and can most likely be methodless and renamed to IRehydratable.
+
+One limitation if we go this route is there isn't a clear place to pass in a flag to include additional properties during serialization and deserialization.
+
+By using the ModelJsonConverter class we can have a place to add additional properties to the JsonSerializerOptions.
+This will allow us to add things like `IgnoreAdditionalProperties` and `Version` to the options without needing to have our own ModelSerializer.
+The `SerializableOptions` would become internal and we would have a converter to convert from `JsonSerializerOptions` + `ModelJsonConverter` to `SerializableOptions`.
+
+### Serialization
+
+```C# Snippet:ModelConverter_Serialize
+DogListProperty dog = new DogListProperty
+{
+    Name = "Doggo",
+    IsHungry = true,
+    Weight = 1.1,
+    FoodConsumed = { "kibble", "egg", "peanut butter" },
+};
+
+JsonSerializerOptions options = new JsonSerializerOptions();
+options.Converters.Add(new ModelJsonConverter());
+
+string json = System.Text.Json.JsonSerializer.Serialize(dog, options);
+```
+
+### Deserialization
+
+```C# Snippet:ModelConverter_Deserialize
+string json = @"[{""LatinName"":""Animalia"",""Weight"":1.1,""Name"":""Doggo"",""IsHungry"":false,""FoodConsumed"":[""kibble"",""egg"",""peanut butter""],""NumberOfLegs"":4}]";
+
+JsonSerializerOptions options = new JsonSerializerOptions();
+options.Converters.Add(new ModelJsonConverter());
+
+DogListProperty dog = System.Text.Json.JsonSerializer.Deserialize<DogListProperty>(json, options);
+```
+
+## Envelope BYOM Case
```

Comment:
This feels like shorthand that people on our team understand that might be hard for users of the interfaces to decode initally.  For public-facing documentation, maybe a different title could help readers?

Reply:
Moved comment to ReadMe PR: https://github.com/Azure/azure-sdk-for-net/pull/37998/files#r1282788963